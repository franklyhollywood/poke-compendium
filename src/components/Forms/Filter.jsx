import React from 'react';

export default function Form({ types, setSelectedType, selectedType }) {
	return (
		<div>
			<select
				aria-label='dropdown'
				value={selectedType}
				onChange={(e) => setSelectedType(e.target.value)}
			>
				<option value='all'>All</option>
				{types.map((type) => (
					<option key={type} value={type}>
						{type}
					</option>
				))}
			</select>
		</div>
	);
}
