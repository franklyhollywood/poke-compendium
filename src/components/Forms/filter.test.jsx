import { screen, render } from '@testing-library/react';
import Filter from './Filter';

it('testing my filter', () => {
	const types = ['fire', 'bug', 'water'];
	render(<Filter types={types} />);
	const dropdown = screen.getAllByLabelText('dropdown');
	expect(dropdown).toMatchSnapshot();
});
