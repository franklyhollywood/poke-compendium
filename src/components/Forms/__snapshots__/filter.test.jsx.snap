// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing my filter 1`] = `
Array [
  <select
    aria-label="dropdown"
  >
    <option
      value="all"
    >
      All
    </option>
    <option
      value="fire"
    >
      fire
    </option>
    <option
      value="bug"
    >
      bug
    </option>
    <option
      value="water"
    >
      water
    </option>
  </select>,
]
`;
